app/controllers/chat_channels_controller.rb

```diff
@@ -184,6 +184,38 @@ def search
     }, ChatChannelSearchSerializer, root: false)
   end
 
+  def archive
+    params.require(:chat_channel_id)
+    params.require(:type)
+
+    topic_type = params[:type] == "newTopic" ? :new : :existing
+    topic_params_ok = \
+      (topic_type == :existing && params[:topic_id].present?) ||
+      (topic_type == :new && params[:title].present?)
+
+    raise Discourse::InvalidParameters if !topic_params_ok
+
+    chat_channel = ChatChannel.find_by(id: params[:chat_channel_id])
```

Comment:
Maybe let's use find_by! here so next `raise` will be not required

Reply:
I think we use `Discourse::NotFound` as a standard though, it's through core a lot and even within the chat controllers this pattern